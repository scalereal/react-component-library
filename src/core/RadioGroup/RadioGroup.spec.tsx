import { render } from '@testing-library/react';
import * as React from 'react';
import { RadioGroup } from '.';

describe('RadioGroup component', () => {
    it('should render with default theme', () => {
        const { container } = render(<RadioGroup />);
        expect(container).toMatchSnapshot();
    });

    it('should render with all the props given', () => {
        const { container } = render(
            <RadioGroup
                display='inline'
                mainLabel="Radio Group Label"
                size="M"
                labelColor='grey'
                bgColor= '#67686b'
                hoverCheckedBgColor='#126AFA'
                checkedBgColor='#126AFA'
                hoverEffect={true}
                labelHoverEffect={true}
                hoverBgColor='#373738'
                hoverLabelColor= '#12fadb'
                errorText='not valid'
                labelPosition='bottom'
            />
            );
        expect(container).toMatchSnapshot();
    });

    it('should render with not-allowed cursor when it is disabled', () => {
        const { container } = render(
            <RadioGroup disabled={true} />
        );
        expect(container).toMatchSnapshot();
    });
});