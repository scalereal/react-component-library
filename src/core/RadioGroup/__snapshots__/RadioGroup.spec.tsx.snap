// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup component should render with all the props given 1`] = `
<div>
  <div
    class="sc-pFZIQ fJhliA"
    display="inline"
  >
    <label
      class="sc-jSgupP cRlLKn"
    >
      not valid
    </label>
    <div
      class="sc-fubCfw fPlukN"
      display="inline"
    >
      <span
        class="sc-bdfBwQ cQNmAO radioGroupLabel"
      >
        Radio Group Label
      </span>
      <ul
        class="sc-iBPRYJ beOvuo"
        display="inline"
      />
    </div>
  </div>
</div>
`;

exports[`RadioGroup component should render with default theme 1`] = `
<div>
  <div
    class="sc-pFZIQ fJhliA"
  >
    <label
      class="sc-jSgupP cRlLKn"
    />
    <div
      class="sc-fubCfw fPlukN"
    >
      <span
        class="sc-bdfBwQ ixTvTj radioGroupLabel"
      />
      <ul
        class="sc-iBPRYJ beOvuo"
      />
    </div>
  </div>
</div>
`;

exports[`RadioGroup component should render with not-allowed cursor when it is disabled 1`] = `
<div>
  <div
    class="sc-pFZIQ fJhliA"
    disabled=""
  >
    <label
      class="sc-jSgupP cRlLKn"
    />
    <div
      class="sc-fubCfw dVTXdZ"
      disabled=""
    >
      <span
        class="sc-bdfBwQ ixTvTj radioGroupLabel"
      />
      <ul
        class="sc-iBPRYJ beOvuo"
        disabled=""
      />
    </div>
  </div>
</div>
`;
